### PR TITLE
Update the lock parameters

### PR DESCRIPTION
**Pull request by iccaszhulili**
After libvirt commit 8dc04cafecd2, we can not config flock and posix_lock parameters any more. Remove them, and we do not need to care about the compatibility, since qemu do not support them from the beginning.

### CODE REVIEW

**Comment by iccaszhulili** [comment]
# avocado run --vt-type libvirt virtual_devices.filesystem_device..cache_mode_auto virtual_devices.filesystem_device..
cache_mode_always                                                                                                                                                                             
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.        
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.                           
JOB ID     : f7d20c4bd48ed29a178c57ab1b43f231e0e40cf7                                                                                                                                         
JOB LOG    : /var/log/avocado/job-results/job-2024-07-23T03.56-f7d20c4/job.log                                                                                                                
 (01/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_on.cache_mode_auto.thread_pool_noset.one_fs.one_guest: ST
ARTED                                                                                                                                                                                         
 (01/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_on.cache_mode_auto.thread_pool_noset.one_fs.one_guest: PA
SS (39.23 s)                                                                                                                                                                                  
 (02/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_off.cache_mode_auto.thread_pool_noset.one_fs.one_guest: S
TARTED                                                                                                                                                                                        
 (02/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_off.cache_mode_auto.thread_pool_noset.one_fs.one_guest: P
ASS (41.00 s)                                                                                                                                                                                 
 (03/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_noset.cache_mode_auto.thread_pool_noset.one_fs.one_guest:
 STARTED
 (03/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_noset.cache_mode_auto.thread_pool_noset.one_fs.one_guest:
 PASS (49.55 s)
 (04/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.hotplug_unplug.detach_device.fs_test.xattr_on.cache_mode_auto.thread_pool_n
oset.one_fs.one_guest: STARTED
 (04/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.hotplug_unplug.detach_device.fs_test.xattr_on.cache_mode_auto.thread_pool_n
oset.one_fs.one_guest: PASS (52.71 s)
 (05/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.hotplug_unplug.detach_device.fs_test.xattr_off.cache_mode_auto.thread_pool_
noset.one_fs.one_guest: STARTED
 (05/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.hotplug_unplug.detach_device.fs_test.xattr_off.cache_mode_auto.thread_pool_
noset.one_fs.one_guest: PASS (47.69 s)
...
 (35/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.memfd_backed.hotplug_unplug.detach_device.fs_test.xattr_off.cache_mode_always.thread_pool_no
set.one_fs.one_guest: STARTED
 (35/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.memfd_backed.hotplug_unplug.detach_device.fs_test.xattr_off.cache_mode_always.thread_pool_no
set.one_fs.one_guest: PASS (55.39 s)
 (36/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.memfd_backed.hotplug_unplug.detach_device.fs_test.xattr_noset.cache_mode_always.thread_pool_
noset.one_fs.one_guest: STARTED
 (36/36) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.memfd_backed.hotplug_unplug.detach_device.fs_test.xattr_noset.cache_mode_always.thread_pool_
noset.one_fs.one_guest: PASS (55.72 s)
RESULTS    : PASS 36 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/log/avocado/job-results/job-2024-07-23T03.56-f7d20c4/results.html
JOB TIME   : 1835.72 s

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
@smitterl FYI.

**Review by yafu-1 — APPROVED** [review]
LGTM

**Review by Yingshun — COMMENTED** [review]
@iccaszhulili  It looks good, but I can't approve it until you fix tp-libvirt scripts. There're too many failures for now without tp-libvirt changes. 

```
 (001/144) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_on.flock_on.lock_posix_off.cache_mode_auto.thread_pool_noset.one_fs.one_guest: ERROR: Attributes values should be str-type of int-type. (5.41 s)
 (002/144) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.hugepages_backed.nop.fs_test.xattr_on.flock_on.lock_posix_on.cache_mode_auto.thread_pool_noset.one_fs.one_guest: ERROR: Attributes values should be str-type of int-type. (5.49 s)

```

**Review by chunfuwen — APPROVED** [review]
lgtm

**Comment by chunfuwen** [comment]
I like to merge this PR as urgent need since it is dependence PR for : https://github.com/autotest/tp-libvirt/pull/5802/files ,which failed s390x gating testing

**Comment by Yingshun** [comment]
@chunfuwen  I understand it's urgent so I'm just working on it. Plz, check with other reviewers who have not approved in the list before merging, thx!

**Pull request merged**
